### ISSUE TITLE
Feature: Flag to disable version checking

### ISSUE THREAD

**Issue by ryaanwells** [mattreid1/firebase_image#4]
Hey 👋 

Thanks for making this library, I've been looking for a good solution that doesn't use `getDownloadURL()` for a while!

My use case has image assets that are permanent - the image binary at a given path will never change. I'm looking to implement a flag that disables the version check logic to save extra calls. Would you be happy if I contribute this?

**Comment by mattreid1** [comment]
Sure if you want! Just open a pull request when you're ready.

I think a named parameter disabling [this line](https://github.com/mattreid1/firebase_image/blob/master/lib/src/cache_manager.dart#L83) should work although I haven't checked for any side effects.

**Comment by ryaanwells** [comment]
Hi @mattreid1, raised in https://github.com/mattreid1/firebase_image/pull/5

I'm thinking there might be a nice way to separate the refresh strategies in a better way (you could use the metadata objects or some pre-defined time interval, for example), but for now this unblocks the immediate use case.

**Comment by ryaanwells** [comment]
Hey @mattreid1, wondering if you'd had a chance to look at this?

In case you missed it there's conversation about an enum approach to support multiple refresh strategies, would be good to have your opinion on.

**Comment by mattreid1** [comment]
Hi, sorry, been working all week.

I think that an enum approach would be good as I agree that if in future we need more options, it would help backwards compatibility.

Right now I think we'll just use two (always and never). I'll make the changes in your fork and then I'll merge the PR. I'll try to merge it in by the end of today. Thank you for your contribution!

**Comment by ryaanwells** [comment]
Hey Matt,

Happy to help out on this (I don't think you'll have write permissions on my branch?). One thing to consider here is that the refresh behaviour is a combination of two dimensions: frequency of checking and what value you use for a version number. The current behaviour is with frequency of "every time the image is used" and version number of "object metadata timestamp". It could be worth making this distinction more explicit to allow for more flexible caching options (e.g. still use the metadata lookup but only do it once every 7 days, or use a local version timestamp, like first download, and check every time the image is loaded until 7 days have passed, etc).

I've suggested using `BY_METADATA_DATE` for the current cache behaviour and I've updated the existing PR to use an enum.

Thanks for the review!
